core/src/main/java/org/jdbi/v3/core/argument/BeanPropertyArguments.java

```diff
@@ -59,6 +61,12 @@
         })
         .build();
 
+    private static boolean shouldSeeProperty(PropertyDescriptor pd) {
+        // 'class' isn't really a property
+        final Method read = pd.getReadMethod();
+        return read == null || read.getDeclaringClass() != Object.class;
```

Comment:
I agree we might want to change this, but this mimics the existing behavior best.  For example if you only provide a setter, the BeanIntrospector will find a property with a null getter.  I experimented by changing this as you suggest and it breaks some existing tests, notably:

```java
    @Test
    public void testBindNonPublicGetter() {
        assertThatThrownBy(() -> new BeanPropertyArguments("foo", new NonPublicGetter()).find("foo.bar", ctx))
            .isInstanceOf(UnableToCreateStatementException.class);
    }
```
we test that having a setter without a getter causes an explosion.  Ignoring non-gettable properties changes this behavior.  If we do accept it as a property here, we later blow up since we can't actually read it, preserving existing behavior.